control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```
@@ -251,11 +254,19 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		r.Log.Info("Cluster infrastructure is still provisioning, will try again later")
 		return r.setAvailableCondition(ctx, hostedControlPlane, oldStatus, hyperv1.ConditionFalse, "WaitingOnInfrastructureReady", "Cluster infrastructure is still provisioning", result, nil)
 	}
-	hostedControlPlane.Status.ControlPlaneEndpoint = hyperv1.APIEndpoint{
-		Host: infraStatus.APIAddress,
-		Port: APIServerPort,
+	switch hostedControlPlane.Spec.ControlPlaneServiceTypeStrategy {
+	case "NodePort":
+		externalAPIPort, _ := strconv.ParseInt(infraStatus.APIPort, 10, 32)
```

eating an error here

---

corrected!